src/Functions/registerFunctions.cpp

```diff
@@ -1,145 +1,60 @@
+void earlyRegisterFunctionsLogical(FunctionFactory & factory);
 
 void registerFunctions()
 {
+    auto symbols = DB::SymbolIndex::instance()->symbols();
     auto & factory = FunctionFactory::instance();
 
-    registerFunctionsArithmetic(factory);
-    registerFunctionsArray(factory);
-    registerFunctionsTuple(factory);
-    registerFunctionsMakeDate(factory);
-    registerFunctionsMap(factory);
-    registerFunctionsBitmap(factory);
-    registerFunctionsBinaryRepr(factory);
-    registerFunctionsCoding(factory);
-    registerFunctionsCodingUUID(factory);
-    registerFunctionChar(factory);
-    registerFunctionsComparison(factory);
-    registerFunctionsConditional(factory);
-    registerFunctionsConversion(factory);
-    registerFunctionCastOrDefault(factory);
-    registerFunctionsDateTime(factory);
-    registerFunctionsEmbeddedDictionaries(factory);
-    registerFunctionsExternalDictionaries(factory);
-    registerFunctionsExternalModels(factory);
-    registerFunctionsFormatting(factory);
-    registerFunctionsHashing(factory);
-    registerFunctionsHigherOrder(factory);
-    registerFunctionsLogical(factory);
-    registerFunctionsMiscellaneous(factory);
-    registerFunctionsRandom(factory);
-    registerFunctionsReinterpret(factory);
-    registerFunctionsRound(factory);
-    registerFunctionsString(factory);
-    registerFunctionsStringArray(factory);
-    registerFunctionsStringSearch(factory);
-    registerFunctionsStringRegexp(factory);
-    registerFunctionsStringSimilarity(factory);
-    registerFunctionsStringTokenExtractor(factory);
-    registerFunctionsURL(factory);
-    registerFunctionsVisitParam(factory);
-    registerFunctionsMath(factory);
-    registerFunctionsGeo(factory);
-    registerFunctionsNull(factory);
-    registerFunctionsJSON(factory);
-    registerFunctionsSQLJSON(factory);
-    registerFunctionToJSONString(factory);
-    registerFunctionsIntrospection(factory);
-    registerFunctionsConsistentHashing(factory);
-    registerFunctionsUnixTimestamp64(factory);
-    registerFunctionBitHammingDistance(factory);
-    registerFunctionTupleHammingDistance(factory);
-    registerFunctionsStringHash(factory);
-    registerFunctionValidateNestedArraySizes(factory);
-    registerFunctionsSnowflake(factory);
-    registerFunctionsTimeWindow(factory);
-    registerFunctionToBool(factory);
-    registerFunctionMinSampleSize(factory);
-
-    //meilisearch
-    registerFunctionMeiliMatch(factory);
-
-#if USE_SSL
-    registerFunctionEncrypt(factory);
-    registerFunctionDecrypt(factory);
-    registerFunctionAESEncryptMysql(factory);
-    registerFunctionAESDecryptMysql(factory);
-    registerFunctionShowCertificate(factory);
-#endif
-    registerFunctionTid(factory);
-    registerFunctionLogTrace(factory);
-    registerFunctionHashID(factory);
+    // These two functions are not build with clickhouse_function,
+    // need to be registered separately
+    earlyRegisterFunctionReplicate(factory);
```

Comment:
Why do we have such inconsistency?

Reply:
These two files are linked to libdbms. The registerXXX() functions will be stripped when linking to the final binary. Unless we change libdbms to OBJECT too.

Reply:
The exception replicate.cpp and FunctionsLogical.cpp, explicit calling for registration for all other functions are removed.
```
list (APPEND dbms_sources Functions/IFunction.cpp Functions/FunctionFactory.cpp Functions/FunctionHelpers.cpp Functions/extractTimeZoneFromFunctionArguments.cpp Functions/replicate.cpp Functions/FunctionsLogical.cpp)
```